### PR TITLE
add jekyll-pwa-plugin

### PR DESCRIPTION
**Pull request by xiaoiver**

### CODE REVIEW

File: docs/_docs/plugins.md [hunk 1]
```diff
@@ -960,6 +960,7 @@ You can find a few useful plugins at the following locations:
 - [Jekyll-Post](https://github.com/robcrocombe/jekyll-post): A CLI tool to easily draft, edit, and publish Jekyll posts.
 - [jekyll-numbered-headings](https://github.com/muratayusuke/jekyll-numbered-headings): Adds ordered number to headings.
 - [jekyll-pre-commit](https://github.com/mpchadwick/jekyll-pre-commit): A framework for running checks against your posts using a git pre-commit hook before you publish them.
+- [jekyll-pwa-plugin](https://github.com/lavas-project/jekyll-pwa): A plugin provides PWA support for Jekyll. Generate a service worker and provides precache and runtime cache with Google Workbox.
```
Comment (mattr-) [inline docs/_docs/plugins.md]: Can you change `Generate` to `Generates` please?

I'm still learning about PWAs but I wasn't aware that there was a dependency there on Google Workbox. Is that just an implementation detail of your plugin?

Comment (DirtyF) [reply docs/_docs/plugins.md]: @mattr- It's an implementation by Google to make it easier to get started.

Comment (xiaoiver) [reply docs/_docs/plugins.md]: Thanks guys. As DirtyF says, Google Workbox has already developed a series of [tools](https://developers.google.com/web/tools/workbox/) . If you use Webpack or Gulp as your build tool, you can easily generate a service worker. But in my blog, I don't want to use even npm so I try to integrate this function in Jekyll build process.

**Comment by ashmaroli** [comment]
@xiaoiver You might want to update the plugin's **usage section**. When i initially tried adding the gem and config block to my test site, `bundle exec jekyll serve` failed with an error..
(resumed properly after I manually copied the contents of your `vendor/` to my site)

**Comment by xiaoiver** [comment]
Hi @ashmaroli , in this plugin I want to copy two files(a polyfill & workbox) under `/vendor` to `_site/js`:
```ruby
Dir.glob(File.expand_path('../vendor/**/*', __FILE__)) do |filepath_to_copy|
    basename = File.basename(filepath_to_copy)
    FileUtils.copy_file(filepath_to_copy, File.join(@site.dest, 'js', basename))
```

So in service worker generated, I can import these scripts correctly: 
`importScripts('#{@site.baseurl}/js/#{vendor_file}');`

Maybe I make a wrong assumption that users put there scripts in **/js**.

**Comment by ashmaroli** [comment]
@xiaoiver  opened an issue ticket at your repo.. best, we continue there..

**Comment by DirtyF** [comment]
@jekyllbot: merge +site

**Pull request merged**